ingestion/src/metadata/ingestion/api/distributed.py

```diff
@@ -0,0 +1,233 @@
+
+    The pattern is:
+    1. Discovery phase: Call discover_entities() to get list of EntityDescriptors
+    2. Processing phase: For each descriptor, call process_entity() in separate pod
+    """
+
+    @abstractmethod
+    def discover_entities(self, entity_type: str) -> List[EntityDescriptor]:
+        """
+        Lightweight discovery of entities to process.
+
+        This should be FAST - only fetch IDs and minimal metadata.
+        Do NOT fetch full entity details or perform expensive operations.
+
+        Args:
+            entity_type: Type of entities to discover (e.g., "table", "dashboard")
+
+        Returns:
+            List of EntityDescriptor with just enough info to process later
+
+        Example for database source:
+            ```python
+            def discover_entities(self, entity_type: str):
+                if entity_type == "table":
+                    entities = []
+                    for schema in self.get_database_schema_names():
+                        for table_name in self.inspector.get_table_names(schema):
+                            entities.append(EntityDescriptor(
+                                id=f"{schema}.{table_name}",
+                                type="table",
+                                metadata={"schema": schema, "table": table_name}
+                            ))
+                    return entities
+            ```
+        """
+        pass
+
+    @abstractmethod
+    def process_entity(self, descriptor: EntityDescriptor) -> Iterable[Either[Any]]:
+        """
+        Process a single entity and yield creation requests.
+
+        This should be STATELESS - can run in any pod with no shared state.
+        All information needed to process must be in the descriptor.
+
+        Args:
+            descriptor: Entity descriptor from discover_entities()
+
+        Yields:
+            Either objects containing entity creation requests or errors
+
+        Example:
+            ```python
+            def process_entity(self, descriptor: EntityDescriptor):
+                if descriptor.type == "table":
+                    table_name = descriptor.metadata["table"]
+                    schema = descriptor.metadata["schema"]
+
+                    # Fetch full table metadata (columns, constraints, etc.)
+                    columns = self.inspector.get_columns(table_name, schema)
+                    pk = self.inspector.get_pk_constraint(table_name, schema)
+
+                    # Create request
+                    yield Either(right=CreateTableRequest(...))
+            ```
+        """
+        pass
+
+    @abstractmethod
+    def get_parallelizable_entity_types(self) -> List[str]:
+        """
+        Return entity types that can be processed in parallel.
+
+        Returns:
+            List of entity type names
+
+        Examples:
+            - Database source: ["table", "stored_procedure", "database_schema"]
```

Comment:
Can parralelize this way ? For example I can't process tables before I processed the schema the tables belong to, same thing with charts, etc.